app/assets/javascripts/application.js

```diff
@@ -24,3 +24,23 @@ $(document).on('turbolinks:load', function() {
   $('.modal-trigger').modal('open');
   $('.modal-trigger').modal('close');
 });
+
+$(document).on('submit', '#login-form', '#signup-form', function(){
+  var $form, $btn;
+
+  $form = $(this);
+  $btn = $form.find('input[type="submit"]');
+  $form.find('.errors').remove();
+
+  $.post({
+    url: this.action,
+    data: $form.serialize()
+  }).error(function(jqXHR, textStatus, errorThrown){
+    $form.prepend('<div class="errors">Invalid email or password</div>');
```

Comment:
We should add to an already created 'errors' div instead of creating a whole new div here. Also we should use the Error that we get from the backend instead of hardcoding 'Invalid email or password' (eventhough it will be this most of the time)